Allow for standard math notation in quantifiers

jim-portegies opened the pull request:
Allows for syntax such as

```
Goal ∀ x ∈ [a, b], True.
Proof.
We need to show that (∀ x ∈ [a, b], True).
Take x ∈ [a, b].
```

Syntax for `x > y` and `x ≥ y` is also supported.

There are also some changes how we send messages to the user. We now send on two channels, as info and notice, so one can end up in the goal panel, and the other ends up as a diagnostic in the editor.

jellooo038 commented:
Please add support for:
 + conventional function notation (if possible) f : X -> Y
 + conditions given using < and <=

jim-portegies commented:
I have refactored the code so we can use `forall` and `exists` notation with arbitrary predicates. I have added the possibilities for quantifiers with conditions `<` and `<=`. I have also reverted back to conventional notation for functions.

jellooo038 reviewed (approved):
Very nice, looks good! 👍 👍
